From 006fba44e9a6fd264963b2aaa6d32c093a197a93 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Tue, 30 Sep 2025 12:08:11 +0200
Subject: [PATCH] base-files: add a turnoff command to the led script

Adds turnoff, turnon, and blink commands to LED init script, plus UCI option to disable LEDs at boot for stealth deployments.

Signed-off-by: John Crispin <john@phrozen.org>
---
 package/base-files/files/etc/init.d/led | 29 +++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/package/base-files/files/etc/init.d/led b/package/base-files/files/etc/init.d/led
index 7f05254c2b..2b171dc71c 100755
--- a/package/base-files/files/etc/init.d/led
+++ b/package/base-files/files/etc/init.d/led
@@ -36,6 +36,10 @@ led_color_set() {
 	echo "${multi_intensity}" > /sys/class/leds/${sysfs}/multi_intensity
 }
 
+extra_command "turnon" "Put the LEDs into their default state"
+extra_command "turnoff" "Turn all LEDs off"
+extra_command "blink" "Blink all LEDs"
+
 load_led() {
 	local name
 	local sysfs
@@ -168,7 +172,32 @@ load_led() {
 	}
 }
 
+turnoff() {
+	for led in `ls /sys/class/leds/`; do
+		echo none > /sys/class/leds/$led/trigger
+		echo 0 > /sys/class/leds/$led/brightness
+	done
+}
+
+turnon() {
+	turnoff
+	. /etc/diag.sh
+	set_state done
+	start
+}
+
+blink() {
+	for led in `ls /sys/class/leds/`; do
+		echo 0 > /sys/class/leds/$led/brightness
+		echo timer > /sys/class/leds/$led/trigger
+	done
+}
+
 start() {
+	[ "$(uci -q get system.@system[-1].leds_off)" = '1' ] && {
+		turnoff
+		exit 0
+	}
 	[ -e /sys/class/leds/ ] && {
 		[ -s /var/run/led.state ] && {
 			local led trigger brightness color
-- 
2.34.1

